### DISCUSSION TITLE
`memory_profiler` not recording timestamps

### DISCUSSION THREAD

**Discussion by ramonemiliani93** [Project-MONAI/MONAI#6745] [Q&A] [answered]
Hi, I am trying to profile memory on a code I have. When using [memory_profiler](https://github.com/pythonprofilers/memory_profiler) if Monai is imported the timestamps are not recorded. There is a warning in the `memory_profiler` library:
```
If your Python file imports the memory profiler from memory_profiler import profile these timestamps will not be recorded. Comment out the import, leave your functions decorated, and re-run.
```
To reproduce, create a file named `example.py`:

 ```python
 #import monai

@profile
def my_func():
    a = [1] * (10**6)
    b = [2] * (2 * 10**7)
    del b
    return a


if __name__ == "__main__":
    my_func()

 ```
To run:
```bash
mprof run --python example.py
mprof plot
```
Timestamps will be recorded depending if Monai is commented out or not. Is Monai using the `memory_profiler` somewhere? or what could be the cause that the timestamps are not recorded.

**Comment by KumoLiu** [comment, marked as answer]
Hi @ramonemiliani93, I can't reproduce the error in "projectmonai/monai:1.2.0" image.
Could you please try the same code in this image?

Thanks!

**Comment by ramonemiliani93** [comment]
@KumoLiu Thanks for the help! Indeed I tried with the image 1.2.0 and 0.8.1 which is my installed version and everything works out. Not sure what could be the problem on my side, I'll try to debug a bit more to understand what is happening.
